src/lib.rs

```diff
@@ -827,23 +835,93 @@ impl SyncClient {
         }
     }
 
+    async fn request_wildcard_handler(
+        wrapped_client: Arc<Mutex<Self>>,
+        mut subscription_rx: MpscReceiver<Msg>,
+    ) {
+        let disconnecting = Self::disconnecting(Arc::clone(&wrapped_client));
+        pin_mut!(disconnecting);
+        loop {
+            // Select between the next message and disconnecting
+            let msg = match future::select(subscription_rx.next(), disconnecting).await {
+                Either::Left((Some(msg), unresolved_disconnecting)) => {
+                    disconnecting = unresolved_disconnecting;
+                    msg
+                }
+                Either::Left((None, _)) => break,
+                Either::Right(((), _)) => break,
+            };
+
+            let mut client = wrapped_client.lock().await;
+            let response_inbox = msg.subject().to_string();
+            if let Some(mut requester_tx) = client.request_inbox_mapping.remove(&response_inbox) {
+                requester_tx.send(msg).await.unwrap_or_else(|err| {
+                    warn!("Could not write response to pending request via mapping channel. Skipping! Err: {}", err);
```

Comment:
Personal preference, but this feels more like an error to me. There is no way we can recover and end up successfully sending the message, correct? In fact, this should really never do you mind adding a `debug_assert!(false);` after the message to indicate such.